### ISSUE TITLE
 libthnets.so is missing

### ISSUE THREAD

**Issue by Rich700000000000** [teradeep/demo-apps#9]
I'm trying to run this package, and it's giving me two different errors.

When I try to run `generic-pc`, it gives me:

    Number of threads used:	8
    qlua: ./frame.lua:63: module 'linuxcamera' not found:
	    no field package.preload['linuxcamera']
	    no file '/home/dave/.luarocks/share/lua/5.1/linuxcamera.lua'
	    no file '/home/dave/.luarocks/share/lua/5.1/linuxcamera/init.lua'
	    no file '/home/dave/torch/install/share/lua/5.1/linuxcamera.lua'
	    no file '/home/dave/torch/install/share/lua/5.1/linuxcamera/init.lua'
	    no file './linuxcamera.lua'
	    no file '/home/dave/torch/install/share/luajit-2.1.0-alpha/linuxcamera.lua'
	    no file '/usr/local/share/lua/5.1/linuxcamera.lua'
	    no file '/usr/local/share/lua/5.1/linuxcamera/init.lua'
	    no file '/home/dave/.luarocks/lib/lua/5.1/linuxcamera.so'
	    no file '/home/dave/torch/install/lib/lua/5.1/linuxcamera.so'
	    no file './linuxcamera.so'
	    no file '/usr/local/lib/lua/5.1/linuxcamera.so'
	    no file '/usr/local/lib/lua/5.1/loadall.so'
    stack traceback:
	    [C]: at 0x7fcb5d5a1970
	    [C]: in function 'require'
	    ./frame.lua:63: in function 'prep_lua_linuxcamera'
	    ./frame.lua:86: in function 'init'
	    run.lua:102: in main chunk

And when I try to run the `opencv` and `generic-embedded` folders, it gives me:

    : error while loading shared libraries: libthnets.so: cannot open shared object file: No such file or directory

How should I go about fixing this?

**Comment by mvitez** [comment]
Have you compiled and installed linuxcamera? The Makefile supposes that 
Torch is in /usr/local, but yours is probably in $HOME/torch/install, so 
copy linuxcamera.so in $(HOME)/torch/install/lib/lua/5.1/.

Regarding thnets, have you installed it? Besides installing it, you have 
to give this each time (or put it in your .bashrc):

export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH

**Comment by Rich700000000000** [comment]
>so copy linuxcamera.so in $(HOME)/torch/install/lib/lua/5.1/.

Ok, I did that, and it WORKS! Thank you so, so much. 

One, down two more to go.

>Regarding thnets, have you installed it? Besides installing it, you have
to give this each time (or put it in your .bashrc):

>export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH

Ok, that didn't work my copy of libthnets.so is in 

`/home/dave/Packages/thnets-master/libthnets.so`, that's where I compiled it. 

What should I add to my .bashrc, bash has always given me trouble.

**Comment by mvitez** [comment]
> |/home/dave/Packages/thnets-master/libthnets.so|, that's where I 
> compiled it.
make install (as root) or
sudo make install (on Debian, Ubuntu, etc...)
inside /home/dave/Packages/thnets-master will copy it in /usr/local/lib 
where it should be.

> What should I add to my .bashrc, bash has always given me trouble.
Add this to .bashrc:

export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH

and then restart the terminal.

**Comment by Rich700000000000** [comment]
Ok, so the 

    : error while loading shared libraries: libthnets.so: cannot open shared object file: No such file or directory

error is fixed, but NOW it's saying it can't find the categories.txt file, and it's IN the same folder. Why is this happening?

**Comment by mvitez** [comment]
It's in the same folder of what? It has to be in the same folder of 
model.net.

**Comment by Rich700000000000** [comment]
Everything is in the same folder:

`dave@dave-HP-EliteBook-8560w:~/Code/ML - Downloaded/demo-apps-master/opencv$ ls`
`8cc.c  8cc.o  aux.t7  categories.txt  classes.t7  icon.jpg  LICENSE.md  Makefile  model.net  README.md  stat.t7  test  thnetsopencvdemo  thnetsopencvdemo.cpp  thnetsopencvdemo.o`

**Comment by mvitez** [comment]
How are you running the program?

You should give the path to that folder after -m, e.g.:

dave@dave-HP-EliteBook-8560w:~/Code/ML - 
Downloaded/demo-apps-master/opencv$ ./thnetsopencvdemo -m .

**Comment by Rich700000000000** [comment]
Good News: It finally Works! Thank you so, so much for all your help.

Bad news: It keeps crashing, almost instantly:

    dave@dave-HP-EliteBook-8560w:~/Code/ML - Downloaded/demo-apps-master/opencv$ ./thnetsopencvdemo -m .
    Segmentation fault (core dumped)
    dave@dave-HP-EliteBook-8560w:~/Code/ML - Downloaded/demo-apps-master/opencv$ ./thnetsopencvdemo -m .

    (thnets opencv demo:3001): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
    Segmentation fault (core dumped)
    dave@dave-HP-EliteBook-8560w:~/Code/ML - Downloaded/demo-apps-master/opencv$ ./thnetsopencvdemo -m .

    (thnets opencv demo:6534): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
    Segmentation fault (core dumped)


What is a "`Segmentation fault`" and how can I stop it?

**Comment by mvitez** [comment]
It means that there's a problem with the software. I cannot help you with this, it requires debugging with a debugger. It's maybe specific to your version of opencv or Qt. Does the generic-embedded demo work?

**Comment by Rich700000000000** [comment]
Yeah, the generic-embedded works fine. It must just be a problem with opencv. Odd.

Anyway, thanks so much for your help. I never would have figured that out, it's been giving me problems for months.
